packages/protocol/thread/analysis.ts

```diff
@@ -0,0 +1,92 @@
+import { AnalysisEntry, Location, PointDescription } from "@replayio/protocol";
+import { AnalysisParams } from "../analysisManager";
+import { gAnalysisCallbacks, sendMessage } from "../socket";
+
+export interface Analysis {
+  analysisId: string;
+  addLocation: (location: Location) => Promise<void>;
```

Comment:
For posterity, I'm going to note that soon `addPoints` will probably join this type.